### PR TITLE
Add Resave Articles Worker

### PR DESCRIPTION
**Pull request by elpic**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Change Resave Articles job to worker, and add missing specs on the user
model when changing the trigger attributes

## Related Tickets & Documents

#5305 

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

### CODE REVIEW

File: spec/workers/users/resave_articles_worker_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+require "rails_helper"
+
+RSpec.describe Users::ResaveArticlesWorker, type: :worker do
+  include_examples "#enqueues_on_correct_queue", "medium_priority", 1
+
+  describe "#perform" do
+    let(:worker) { subject }
+
+    context "with comment" do
```
Comment (mstruve) [inline spec/workers/users/resave_articles_worker_spec.rb]: I think you mean "with user" here and "without user" below 😉

Been there, done that before! hehe

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -507,6 +507,125 @@ def user_from_authorization_service(service_name, signed_in_resource, cta_varian
         end.not_to have_enqueued_job(Users::SubscribeToMailchimpNewsletterJob).exactly(:once).with(user.id)
       end
     end
+
+    describe "#conditionally_resave_articles" do
```
Comment (mstruve) [inline spec/models/user_spec.rb]: HOLY AWESOME specs!!!! 🙌

File: spec/workers/users/resave_articles_worker_spec.rb (hunk 1 above)
Comment (elpic) [reply spec/workers/users/resave_articles_worker_spec.rb]: 😊

**Comment by elpic** [comment]
Sure @mstruve I will take a look at those

**Comment by elpic** [comment]
Hello @mstruve, is there a way to re-trigger the tests, the tests that are failing doesn't seem to be related to my change.

**Comment by rhymes** [comment]
Hi @elpic, I think you are having a conflict right now. Once you solved it we'll keep an eye on the specs. Sorry for the back and forth

**Comment by mstruve** [comment]
Fixed the conflict, we had another contributor fix that spec and their PR was merged just ahead of yours. Everything should be good to go now!

**Pull request merged**